zebrad/tests/common/get_block_template_rpcs/get_block_template.rs

```diff
@@ -23,6 +36,11 @@ use crate::common::{
 /// We've seen it take anywhere from 1-45 seconds for the mempool to have some transactions in it.
 pub const EXPECTED_MEMPOOL_TRANSACTION_TIME: Duration = Duration::from_secs(45);
 
+/// Number of times to retry a block proposal request with the template returned by the getblocktemplate RPC.
+///
+/// We've seen spurious rejections of block proposals.
```

Comment:
This seems like it could be a bug in the block proposal validation code or the test harness.

Can you show me one of these rejections?

Reply:
I'll add more detailed reject-reasons for debugging this and report back. So far, I observed one rejection from before the state request but after the merkle root validity check, and one from the merkle root validity check failing.

Reply:
The transactions weren't sorted when generating the default Merkle root 🤷